netlink-request/src/lib.rs

```diff
@@ -143,8 +143,8 @@ mod linux {
                     .map_err(|e| io::Error::new(io::ErrorKind::InvalidData, e))?;
                 match response.payload {
                     // We've parsed all parts of the response and can leave the loop.
-                    NetlinkPayload::Ack(_) | NetlinkPayload::Done => return Ok(responses),
-                    NetlinkPayload::Error(e) => return Err(e.into()),
+                    NetlinkPayload::Done(_) => return Ok(responses),
+                    NetlinkPayload::Error(e) if e.code.is_some() => return Err(e.into()),
```

Comment:
Looking at the changes upstream, I *think* this would be logically the same. I'm pretty sure we still want to recognize a zero-error-code `Ack` (i.e. not a `Nack`) as valid.

Reply:
> I'm pretty sure we still want to recognize a zero-error-code `Ack` (i.e. not a `Nack`) as valid.

Alright, we already do. But the question is whether we should exit to loop in this case, or wait for `Done`?

Reply:
Oh I just realized your comment has a code suggestion. It makes sense to me now, let me iterate the code.